ui/server.py

```diff
@@ -292,8 +358,56 @@ def filter(self, record: logging.LogRecord) -> bool:
+
+async def check_status(): # Task to Validate user config shortly after startup.
+    # Check that the loaded config.json yielded a server in a known valid state.
+    # When issues are found, try to fix them when possible and warn the user.
+    device_count = 0
+    # Wait for devices to register and/or change names.
+    THREAD_START_DELAY = 5 # seconds - Give time for devices/threads to start.
+    for i in range(10): # Maximum number of retry.
+        await asyncio.sleep(THREAD_START_DELAY)
+        new_count = task_manager.is_alive()
+        # Stops retry once no more devices show up.
+        if new_count > 0 and device_count == new_count: break
+        device_count = new_count
+
+    if 'render_devices' in config and task_manager.is_alive() <= 0: # No running devices, probably invalid user config. Try to apply defaults.
+        print('WARNING: No active render devices after loading config. Validate "render_devices" in config.json')
+        task_manager.start_render_thread('auto') # Detect best device for renders
+        task_manager.start_render_thread('cpu') # Allow CPU to be used for renders
+        await asyncio.sleep(THREAD_START_DELAY) # delay message after thread start.
+        print('Default render devices loaded to replace missing render_devices', config['render_devices'])
+
+    display_warning = False
+    if not 'render_devices' in config and task_manager.is_alive(0) <= 0: # No config set, is on auto mode and without cuda:0
+        task_manager.start_render_thread('cuda') # An other cuda device is better and cuda:0 is missing, start it...
+        display_warning = True # And warn user to update settings...
+        await asyncio.sleep(THREAD_START_DELAY) # delay message after thread start.
+
+    if display_warning or task_manager.is_alive(0) <= 0:
+        print('WARNING: GFPGANer only works on GPU:0, use CUDA_VISIBLE_DEVICES if GFPGANer is needed on a specific GPU.')
+        print('Using CUDA_VISIBLE_DEVICES will remap the selected devices starting at GPU:0 fixing GFPGANer')
+        print('Add the line "@set CUDA_VISIBLE_DEVICES=N" where N is the GPUs to use to config.bat')
+        print('Add the line "CUDA_VISIBLE_DEVICES=N" where N is the GPUs to use to config.sh')
+
+# Start the task_manager
+task_manager.default_model_to_load = resolve_model_to_use()
+if 'render_devices' in config: # Start a new thread for each device.
+    if isinstance(config['render_devices'], str):
+        config['render_devices'] = config['render_devices'].split(',')
+    if not isinstance(config['render_devices'], list):
+        raise Exception('Invalid render_devices value in config.')
+    for device in config['render_devices']:
+        task_manager.start_render_thread(device)
+else:
+    # Select best GPU device using free memory, if more than one device.
+    task_manager.start_render_thread('auto') # Detect best device for renders
+    task_manager.start_render_thread('cpu') # Allow CPU to be used for renders
```

Comment:
Is there any reason, why you start loading the model to gpu and cpu for "new" users without render_devices config?

Reply:
It's allowing them to be used, but preload checks and will only work on gpu. Also tonight a change is coming to allow preload on cpu when no gpu are present. But at the moment preload is not applied on cpu.

Reply:
I've pinged @madrang regarding this question

Reply:
But Cmdr2 did change it ti always preload...
https://github.com/cmdr2/stable-diffusion-ui/pull/404/commits/389e3397ecc8b2e47217d0428a8291d0c5253e9b

Reply:
Thanks @rbertus2000 for pointing this out! This bit will undergo some changes soon